exe/proteus

```diff
@@ -74,79 +81,252 @@ command :info do |c|
   end
 end
 
-desc 'Search DNS record(s) (do not include domain!)'
-command :search do |c|
-  c.desc 'searches for a host record'
-  c.arg_name 'keyword'
-  c.command :host do |a|
-    a.action do |_global_options, _options, args|
+#####
+# Actions specific to Host (A) records
+desc 'Performs actions on host records'
+command :host do |h|
+  h.desc 'Search for a host record (do not include domain!)'
+  h.arg_name 'keyword'
+  h.command :search do |s|
+    s.action do |_global_options, _options, args|
       validate_argnum(1, args.length)
       ap @proteus.search_by_object_types(args[0], Proteus::Types::HOSTRECORD)
     end
   end
 
-  c.desc 'searches for an alias record'
-  c.arg_name 'keyword'
-  c.command :alias do |a|
+  h.desc 'Add a host record (A)'
+  h.arg_name 'fqdn ip'
+  h.command :add do |a|
     a.action do |_global_options, _options, args|
-      validate_argnum(1, args.length)
-      ap @proteus.search_by_object_types(args[0], Proteus::Types::CNAMERECORD)
+      validate_argnum(2, args.length)
+      ap @proteus.add_host_record(args[0], args[1])
     end
   end
 
-  c.desc 'searches for an external record'
-  c.arg_name 'keyword'
-  c.command :external do |a|
-    a.action do |_global_options, _options, args|
+  h.desc 'Show a host record'
+  h.arg_name 'fqdn'
+  h.command :show do |s|
+    s.action do |_global_options, _options, args|
       validate_argnum(1, args.length)
-      ap @proteus.search_by_object_types(args[0], Proteus::Types::EXTERNALHOST)
+      config_id = @proteus.get_entities.first.id
+      collect_pairs(args[0].split '.').each do |name, hint|
+        unless name == '' || hint == ''
+          # TODO: paging is required here -- probably need a helper
+          @proteus.get_zones_by_hint(config_id, 0, 10, "hint=#{hint}").each do |z|
+            begin
+              puts @proteus.get_entity_by_name(z.id, name, Proteus::Types::HOSTRECORD)
+            rescue Proteus::ApiEntityError::EntityNotFound
+              puts "#{name} not found in zone #{hint} with id #{z.id}"
+            end
+          end
+        end
+      end
```

Comment:
lol yeah i had the same reaction, its mostly the GLI DSL though that's causing it....